src/uct/ib/base/ib_log.c

```diff
@@ -39,16 +39,19 @@ void uct_ib_log_dump_sg_list(uct_ib_iface_t *iface, uct_am_trace_type_t type,
                      sg_list[i].addr, sg_list[i].length, sg_list[i].lkey);
         }
 
-        len = ucs_min(sg_list[i].length, (void*)data + sizeof(data) - md);
-        memcpy(md, (void*)sg_list[i].addr, len);
-
         s               += strlen(s);
-        md              += len;
-        total_len       += len;
-        total_valid_len += sg_list[i].length;
+
+        if (data_dump) {
```

Comment:
is this if still needed?

Reply:
yes,  for get(RDMA_READ) it passed down NULL dump func, but it is trying to copy the data (GPU memory) to buffer